### PR TITLE
Recaptcha

### PR DESCRIPTION
**Pull request by joffe97**
* Created recaptcha hook
* Added recaptcha to giver form
* Created recaptcha controller and dto
* Added recaptcha secret to github secrets
* Recaptcha only runs on prod
* Updated some packages

Note: This has to be tested live. The github action and method for getting the secret in the controller has not been tested.

### CODE REVIEW

File: GiEnJul/ClientApp/.env.production [hunk 1]
```diff
@@ -1,3 +1,5 @@
 REACT_APP_DEV_TENANT_AUTH0=dev-r7fmessb.eu.auth0.com
 REACT_APP_DEV_CLIENTID_AUTH0=sxpa9h7p3UnlhP2aIttZHlLVr41rU0RR
-REACT_APP_API_URL=https://gienjultestapi.azurewebsites.net/api
\ No newline at end of file
+REACT_APP_API_URL=https://gienjultestapi.azurewebsites.net/api
+REACT_APP_RECAPTCHA_SITE_KEY=6Lc9AsobAAAAACGIbQN8_OT68cHhlkWsT1xCwHl3
```
Comment (RonHaland) [inline GiEnJul/ClientApp/.env.production:4]: uhhhh?

File: GiEnJul/ClientApp/.env [hunk 2]
```diff
@@ -1,3 +1,5 @@
 REACT_APP_DEV_TENANT_AUTH0=dev-r7fmessb.eu.auth0.com
 REACT_APP_DEV_CLIENTID_AUTH0=sxpa9h7p3UnlhP2aIttZHlLVr41rU0RR
-REACT_APP_API_URL=https://localhost:5001/api
\ No newline at end of file
+REACT_APP_API_URL=https://localhost:5001/api
+REACT_APP_RECAPTCHA_SITE_KEY=6Lc9AsobAAAAACGIbQN8_OT68cHhlkWsT1xCwHl3
```
Comment (RonHaland) [inline GiEnJul/ClientApp/.env:4]: ?

File: .github/workflows/main_backend.yml [hunk 3]
```diff
@@ -2,36 +2,40 @@ name: Build and Test .NET API
...
+    runs-on: ubuntu-latest
 
     steps:
       - name: Checkout
         uses: actions/checkout@v2
 
+      - name: Get recaptcha secret
+        env:
+          RECAPTCHA_SECRET: ${{ secrets.RECAPTCHA_SECRET }}
+
```
Comment (RonHaland) [inline .github/workflows/main_backend.yml:28]: Don't think this does what you think :)

File: GiEnJul/ClientApp/src/components/registerGiver/Summary.tsx [hunk 4]
```diff
@@ -130,152 +152,142 @@ const SummaryRegistration: React.FC<Props> = ({
     <>
```
Comment (joffe97) [inline GiEnJul/ClientApp/src/components/registerGiver/Summary.tsx:152]: This template is unchanged

File: .github/workflows/CI_backend.yml [hunk 5]
```diff
@@ -2,25 +2,25 @@ name: Build and Test .NET API
```
Comment (RonHaland) [inline .github/workflows/CI_backend.yml:2]: I would do 
```
git restore .github/workflows/CI_backend.yml --source=master
```
To restore the file from master, as no changes are actually made here.

File: GiEnJul/Controllers/GiverController.cs [hunk 6]
```diff
@@ -18,20 +19,35 @@ public class GiverController : ControllerBase
...
             _log = log;
             _mapper = mapper;
+            _recaptchaVerifier = recaptchaVerifier;
         }
 
         // POST api/<GiverController>
         [HttpPost]
         public async Task<ActionResult<PostGiverResultDto>> PostAsync([FromBody] PostGiverDto giverDto)
         {
+
```
Comment (RonHaland) [inline GiEnJul/Controllers/GiverController.cs:43]: This newline 🤢

File: GiEnJul/Utilities/RecaptchaVerifier.cs [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    public class RecaptchaVerifier : IRecaptchaVerifier
+    {
+        public RecaptchaVerifier(ISettings settings)
+        {
+            _secret = settings.RecaptchaSecret;
+        }
+
+        private readonly string _secret;
```
Comment (RonHaland) [inline GiEnJul/Utilities/RecaptchaVerifier.cs:22]: usually we have fields above constructors

**Review by RonHaland — APPROVED** [review]
few small suggestions

**Review by RonHaland — APPROVED** [review]

**Pull request merged**